packages/smooth_app/lib/pages/onboarding/country_selector.dart

```diff
@@ -0,0 +1,155 @@
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _initFuture = _init();
+  }
+
+  Future<void> _init() async {
+    final String locale = Localizations.localeOf(context).languageCode;
+    final List<Country> localizedCountries =
+        await IsoCountries.iso_countries_for_locale(locale);
+    _countryList = sanitizeCountriesList(localizedCountries);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    return FutureBuilder<void>(
+        future: _initFuture,
+        builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+          if (snapshot.hasError) {
+            return Text('Fatal Error: ${snapshot.error}');
+          }
+          if (snapshot.connectionState != ConnectionState.done) {
+            return const CircularProgressIndicator();
+          }
+          return Padding(
+            padding:
+                const EdgeInsets.only(top: MEDIUM_SPACE, bottom: LARGE_SPACE),
+            child: DropdownButtonFormField<String>(
+              value: _chosenValue,
+              style: const TextStyle(color: Colors.black),
+              decoration: InputDecoration(
+                enabledBorder: OutlineInputBorder(
+                  borderSide: const BorderSide(
+                      color: Color.fromARGB(255, 235, 235, 235)),
+                  borderRadius: BorderRadius.circular(VERY_LARGE_SPACE),
+                ),
+                filled: true,
+                fillColor: const Color.fromARGB(255, 235, 235, 235),
+              ),
+              items:
+                  _countryList.map<DropdownMenuItem<String>>((Country country) {
+                return DropdownMenuItem<String>(
+                  value: country.name,
+                  child: Text(country.name),
+                );
+              }).toList(),
+              hint: Text(
+                appLocalizations.country_chooser_label,
+                style: const TextStyle(
+                    color: Colors.black,
+                    fontSize: 16,
+                    fontWeight: FontWeight.w600),
+              ),
+              onChanged: (String? value) {
+                setState(() {
+                  if (value != null) {
+                    // TODO(jasmeet): Store pref in _userPreferences
+                    _chosenValue = value;
+                  }
+                });
+              },
+            ),
+          );
+        });
+  }
+
+  List<Country> sanitizeCountriesList(List<Country> localizedCountries) {
+    final List<Country> finalCountriesList = <Country>[];
+    final Map<String, OpenFoodFactsCountry> oFFIsoCodeToCountry =
+        <String, OpenFoodFactsCountry>{};
+    final Map<String, Country> localizedIsoCodeToCountry = <String, Country>{};
+    for (final OpenFoodFactsCountry c in OpenFoodFactsCountry.values) {
+      oFFIsoCodeToCountry.putIfAbsent(c.iso2Code.toLowerCase(), () => c);
```

Comment:
For the record here there's no ambiguity: it will always be absent.
`oFFIsoCodeToCountry[c.iso2Code.toLowerCase()] = c;` is easier to read.